File: Hcs-Platform-App/src/app/models/hcs-basic-model/PlatformGroup.ts
import { ApiEntry, Key, Property } from '../../hcs-lib/ApiEntry';
import { Enums } from './enums';
@ApiEntry('api/entity/Hcs.Platform.BaseModels.PlatformGroup')
export class PlatformGroup {
    @Key()
    public Id: number;
    @Property()
    public Account: string;
    @Property()
    public Name: string;
    @Property()
    public IsEnabled: boolean;
}

File: Hcs.Platform.BaseModels/PlatformGroup.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Hcs.Platform.Abstractions.Role;
using Hcs.Platform.Data;
using Hcs.Platform.User;

namespace Hcs.Platform.BaseModels
{
    public class PlatformGroup : IPlatformEntity
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public bool IsEnabled { get; set; }
        public long? CreatedBy { get; set; }
        public long? LastUpdatedBy { get; set; }
        public DateTime? CreatedTime { get; set; }
        public DateTime? LastUpdatedTime { get; set; }
        IPlatformUser IPlatformEntity.CreatedByUser => CreatedByUser;
        public PlatformUser CreatedByUser { get; set; }
        public PlatformUser LastUpdatedByUser { get; set; }
        IPlatformUser IPlatformEntity.LastUpdatedByUser => LastUpdatedByUser;
        public ICollection<PlatformUserGroup> PlatformUserGroups { get; set; }
        public ICollection<PlatformGroupRole> PlatformGroupRoles { get; set; }
    }
}
